sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategySuite.scala

```diff
@@ -311,6 +311,22 @@ class DataSourceStrategySuite extends PlanTest with SharedSparkSession {
     assert(PushableColumnAndNestedColumn.unapply(Abs('col.int)) === None)
   }
 
+  test("SPARK-36644: Push down boolean column filter") {
+    testTranslateFilter('col.boolean, Some(sources.EqualTo("col", true)))
+
+    val t = "test_table"
+    withTable(t) {
+      import testImplicits._
+      Seq(Some(true), Some(false), None).toDF().write.saveAsTable(t)
+      val df = spark.table(t).where("value")
+      df.queryExecution.executedPlan.collectFirst {
+        case f: FileSourceScanExec =>
+          assert(f.metadata("PushedFilters") == "[IsNotNull(value), EqualTo(value,true)]")
+      }
+      checkAnswer(df, Row(true))
+    }
```

Comment:
hm, do we need e2e test here?

Reply:
I think we can just remove

Reply:
We should follow the existing code in this suite to add a UT, not an end-to-end test

Reply:
Thank you! Removed